llvm/lib/Target/AArch64/AArch64InstrInfo.cpp

```diff
@@ -327,12 +328,27 @@ void AArch64InstrInfo::insertIndirectBranch(MachineBasicBlock &MBB,
       .addImm(16);
 }
 
-// Branch analysis.
+bool AArch64InstrInfo::isCondBranchPredictable(
+    const MachineInstr &CondBr, const MachineLoopInfo &MLI) const {
+  MachineLoop *Loop = MLI.getLoopFor(CondBr.getParent());
+  if (!Loop)
+    return false;
+  return Loop->isLoopInvariant(CondBr, /*ExcludeReg=*/0, /*RecursionDepth=*/2);
```

Comment:
Why is `RecursionDepth=2` here?

Reply:
This way we reproduce the same logic that worked for AArch64 in EarlyIfConversion.

Reply:
Oh I see! Thanks!